### PR TITLE
[WIP] concepts: dvc-project document

### PR DESCRIPTION
**Pull request by iesahin**
This should be merged after #2097. 

This is the DVC Project BC document. 

- Updates the tooltip for DVC project. 
- Adds a concept document for _DVC project_
- Relates to #550 

### TODO

- [ ] Describe nested projects

### CODE REVIEW

**Review by shcheklein — APPROVED** [review]
Looks good to me. The only question I have is either we should include something about DVC subproject (when `dvc init` is run in a subdirectory of a Git project).

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 1]
```diff
@@ -9,10 +9,39 @@ match:
     'DVC repository',
     'DVC repositories',
     repository,
-    repositories,
```
Comment (shcheklein) [inline content/docs/user-guide/basic-concepts/dvc-project.md:12]: we'll need to get it back?

File: content/docs/sidebar.json [hunk 2]
```diff
@@ -127,7 +133,6 @@
           "merge-conflicts"
         ]
       },
-      "experiment-management",
```
Comment (jorgeorpinel) [inline content/docs/sidebar.json:130]: Why is this being deleted?

File: content/docs/sidebar.json [hunk 3]
```diff
@@ -361,17 +366,6 @@
         "label": "run",
         "slug": "run"
       },
-      {
-        "label": "stage",
-        "slug": "stage",
-        "source": "stage/index.md",
```
Comment (jorgeorpinel) [inline content/docs/sidebar.json:367]: Same Q.

File: yarn.lock [hunk 4]
```diff
@@ -13378,11 +13378,16 @@ prettier-linter-helpers@^1.0.0:
   dependencies:
     fast-diff "^1.1.2"
 
-prettier@^2.0.4, prettier@^2.0.5:
+prettier@^2.0.5:
```
Comment (jorgeorpinel) [inline yarn.lock:13381]: #2097 is merged but I still see yarn.lock changes here. May need to merge `master` into this branch?

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 5]
```diff
@@ -12,7 +12,36 @@ match:
     repositories
   ]
 tooltip: >-
-  Initialized by running `dvc init` in the **workspace** (typically a Git
-  repository). It will contain the `.dvc/` directory, as well as `dvc.yaml` and
-  `.dvc` files created with commands such as `dvc add` or `dvc run`.
+  A DVC project is a directory initialized by `dvc init` that contains a `.dvc`
+  directory. It contains all <abbr>metafiles</abbr> related to project and also
+  includes the underlying <abbr>cache</abbr>.
+  [📖](/doc/user-guide/basic-concepts/dvc-project)
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:18]: I think we should avoid changing tooltips in these PRs unless there's a strong reason. Tooltips are currently written to be as short as possible and checked visually. Also, in this case the change states that a project is a directory when that's not necessarily the case conceptually.

Comment (jorgeorpinel) [reply content/docs/user-guide/basic-concepts/dvc-project.md:18]: Adding the 📖 links to tooltips in these PRs is great though.

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 6]
```diff
@@ -12,7 +12,36 @@ match:
...
 ---
+
+# DVC Project
+
+A DVC project (or DVC repository) is a directory initialized by `dvc init`. It
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:23]: > A DVC project (or DVC repository) is a directory

Not sure we want to say it's a project is a directory. That increases the overlap with the workspace concept. Please refer to the discussion we had earlier in https://github.com/iterative/dvc.org/issues/1127#issuecomment-773630307

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 7]
```diff
@@ -12,7 +12,36 @@ match:
...
+contains [`.dvc` directory](/doc/user-guide/project-structure/internal-files),
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:24]: > contains `.dvc` directory

How about something like "includes the configuration (see `.dvc/`) and |cache|, as well as your current |workspace| ..."

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 8]
```diff
@@ -12,7 +12,36 @@ match:
...
+`dvc.yaml` files that describe <abbr>pipelines</abbr> and `*.dvc` files for each
+tracked data file. Files in <abbr>remotes</abbr> and <abbr>cache</abbr>, and
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:26]: > `*.dvc` files for each tracked data file

Data files are tracked by dvc.lock too.

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 9]
```diff
@@ -12,7 +12,36 @@ match:
...
+files like <abbr>metrics</abbr> and <abbr>plots</abbr> produced by DVC are also
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:27]: - I don't think we consider files in remote storage part of the project usually (they relate but I wouldn't mention it). Remotes are a separate concept: a backup and a way to share cached data 
- Agree to emphasize metrics/plots 👍  but lets then also mention parameters I think. 
- However we're missing mentioning general git-tracked files like code?

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 10]
```diff
@@ -12,7 +12,36 @@ match:
...
+considered part of the project.
+
+Although it can used without a standard VCS or with alternative VCS's, a DVC
+project _usually_ has a `.git` directory along with a `.dvc` directory. A DVC
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:31]: Having a .git directory is sort of an "implementation detail". Better to mention that (parts of) the project are usually git-tracked which enables pipelining, experimenting, etc features of DVC?

Comment (jorgeorpinel) [reply content/docs/user-guide/basic-concepts/dvc-project.md:31]: Also we can talk specifically about Git everywhere I think. It's the only supported VSC 🙂

File: content/docs/user-guide/basic-concepts/dvc-project.md [hunk 11]
```diff
@@ -12,7 +12,36 @@ match:
...
+project can be called a DVC repository to emphasize the presence of underlying
+VCS. In most of the cases DVC project and DVC repository are synonyms.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/basic-concepts/dvc-project.md:33]: > In most of the cases DVC project and DVC repository are synonyms.

Hmmm. Not really. Better not to say this? Each term will be used under the appropriate context in the docs.

**Review by jorgeorpinel — COMMENTED** [review]
Here's a review of this concept text. Thanks

**Comment by jorgeorpinel** [comment]
> should include something about DVC subproject?

I think it's better leave that explanation that where it is now: https://dvc.org/doc/command-reference/init#initializing-dvc-in-subdirectories but a mention of "nested projects" could be included (link to that doc#section).

**Comment by jorgeorpinel** [comment]
Closing for now as stale, feel free to reply/reopen.

**Pull request closed without merging**